### PR TITLE
APM-74: NoSQL CouchDB migration

### PR DESCRIPTION
**Pull request by gulabbisht**
https://acquiaps.atlassian.net/browse/APM-74

### CODE REVIEW

File: modules/examples/apm_couch_migrate/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    'database' => 'drupal8couch',
+  ],
+];
+ ```
+* Once DB settings done, import the data from the `data` directory
+of this module to your mongodb database by following steps -
+* Navigate to `http://127.0.0.1:5984/_utils/` and create database `drupal8couch`.
+* Inside the database create 3 documents with _id content, tags and users.
+* Now edit the documents and add json data from the corresponding `data` directory files.
+* Then enable the module `apm_couch_migrate` and run migration.
```
Comment (joshirohit100) [inline modules/examples/apm_couch_migrate/README.md]: this looks italic

File: modules/examples/apm_couch_migrate/src/Plugin/migrate/source/ApmCouch.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate\MigrateException;
+use PHPOnCouch\CouchClient;
+use CouchDB\Model\BSONDocument;
+use CouchDB\Driver\Exception\ConnectionTimeoutException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Source plugin for the CouchDB.
```
Comment (joshirohit100) [inline modules/examples/apm_couch_migrate/src/Plugin/migrate/source/ApmCouch.php]: just add some info/doc about this plugin like required keys `collection`, `field_mapping` etc.

File: modules/examples/apm_couch_migrate/README.md (hunk 1 above)
Comment (gulabbisht) [reply modules/examples/apm_couch_migrate/README.md]: I have tried to update the file but doesn't seem to be working.

**Comment by joshirohit100** [comment]
pr looks fine to me.

cc @amitgoyal for review and merge.

File: modules/examples/apm_couch_migrate/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+$settings['couchdb'] = [
+  // Connection info.
+  'default' => [
+    'dns' => 'http://localhost:5984',
+    'database' => 'drupal8couch',
+  ],
+];
+ ```
+* Once DB settings done, import the data from the `data` directory
+of this module to your mongodb database by following steps:
```
Comment (pranitjha) [inline modules/examples/apm_couch_migrate/README.md]: Replace 'your mongodb database' with 'your couchdb database'

File: modules/examples/apm_couch_migrate/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    * Navigate to `http://127.0.0.1:5984/_utils/` and create database `drupal8couch`.
+    * Inside the database create 3 documents with `_id` content, tags and users.
+    * Now edit the documents and add json data from the corresponding `data` directory files.
```
Comment (pranitjha) [inline modules/examples/apm_couch_migrate/README.md]: Steps to import data to couchDB needs to be added.

File: modules/examples/apm_couch_migrate/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# CouchDB Migrate
+
+This is a sample module to migrate data from CouchDB to Drupal8.
+
+## Requirements
+
+* This module requires php `php-on-couch` library to be installed.
```
Comment (pranitjha) [inline modules/examples/apm_couch_migrate/README.md]: First requirement will be installing couchDB on machine.

File: modules/examples/apm_couch_migrate/README.md (hunk 4 above)
Comment (gulabbisht) [reply modules/examples/apm_couch_migrate/README.md]: @pranitjha above 3 subpoints contains information to import the data from `data` directory of the module.

File: modules/examples/apm_couch_migrate/README.md (hunk 3 above)
Comment (gulabbisht) [reply modules/examples/apm_couch_migrate/README.md]: Updates done in https://github.com/joshirohit100/migration_glue/pull/78

File: modules/examples/apm_couch_migrate/README.md (hunk 5 above)
Comment (gulabbisht) [reply modules/examples/apm_couch_migrate/README.md]: Updates done in https://github.com/joshirohit100/migration_glue/pull/78

**Pull request merged**